### PR TITLE
Downloads weights in examples/stable_diffusion.py

### PR DESCRIPTION
**Pull request by python273**

### CODE REVIEW

**Comment by geohot** [comment]
Very duplicated code with fetch.

**Comment by python273** [comment]
the model is 4GB, `fetch` keeps whole file in memory, so I though it would be better to have a separate function that downloads directly to disk 🤔

File: extra/utils.py [hunk 1]
```diff
@@ -9,21 +9,31 @@ def fetch(url):
...
-    r = requests.get(url, stream=True)
-    assert r.status_code == 200
-    progress_bar = tqdm(total=int(r.headers.get('content-length', 0)), unit='B', unit_scale=True, desc=url)
-    dat = b''.join((x,progress_bar.update(len(x)))[0] for x in r.iter_content(chunk_size=16384))
-    with open(fp+".tmp", "wb") as f:
-      f.write(dat)
-    os.rename(fp+".tmp", fp)
+  if getenv("NOCACHE"):
+    try:
+      os.unlink(fp)
```
Comment (geohot) [inline extra/utils.py:16]: This is different behavior for NOCACHE

Comment (python273) [reply extra/utils.py]: good now?

**Pull request merged**